plugin.video.newson/default.py

```diff
@@ -0,0 +1,270 @@
+#   Copyright (C) 2017 Lunatixz
+#
+#
+# This file is part of NewsOn.
+#
+# NewsOn is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# NewsOn is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NewsOn.  If not, see <http://www.gnu.org/licenses/>.
+
+# -*- coding: utf-8 -*-
+import os, sys, time, datetime, re, traceback, feedparser
+import urllib, urllib2, socket, json, collections
+import xbmc, xbmcgui, xbmcplugin, xbmcvfs, xbmcaddon
+
+from simplecache import SimpleCache
+# Plugin Info
+ADDON_ID      = 'plugin.video.newson'
+REAL_SETTINGS = xbmcaddon.Addon(id=ADDON_ID)
+ADDON_NAME    = REAL_SETTINGS.getAddonInfo('name')
+SETTINGS_LOC  = REAL_SETTINGS.getAddonInfo('profile')
+ADDON_PATH    = REAL_SETTINGS.getAddonInfo('path').decode('utf-8')
+ADDON_VERSION = REAL_SETTINGS.getAddonInfo('version')
+ICON          = REAL_SETTINGS.getAddonInfo('icon')
+FANART        = REAL_SETTINGS.getAddonInfo('fanart')
+NEWSART       = os.path.join(ADDON_PATH,'resources','images','newscast.jpg')
+CLIPART       = os.path.join(ADDON_PATH,'resources','images','videoclips.jpg')
+
+## GLOBALS ##
+TIMEOUT     = 30
+DEBUG       = True
+BASE_URL    = 'http://watchnewson.com/'
+BASE_API    = 'http://watchnewson.com/api/linear/channels'
+USER_REGION = 'US'
+MENU        = [("Newscasts"  , '0', 0, False, {"thumb":NEWSART,"poster":NEWSART,"fanart":FANART,"icon":ICON,"logo":ICON}),
+               ("Video Clips", '2', 2, False, {"thumb":CLIPART,"poster":CLIPART,"fanart":FANART,"icon":ICON,"logo":ICON})]
+
+def log(msg, level=xbmc.LOGDEBUG):
+    if DEBUG == True:
+        if level == xbmc.LOGERROR:
+            msg += ' ,' + traceback.format_exc()
+        xbmc.log(ADDON_ID + '-' + ADDON_VERSION + '-' + stringify(msg), level)
+        
+def stringify(string):
+    if isinstance(string, list):
+        string = stringify(string[0])
+    elif isinstance(string, (int, float, long, complex, bool)):
+        string = str(string) 
+    if isinstance(string, basestring):
+        if not isinstance(string, unicode):
```

Comment:
it's not an if elif else statement. Stringify doesn't cover all possible types(). The if elif attempts to convert to a string. Then it checks to make sure only strings are being evaluated for encoding.